src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -426,6 +461,60 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     }
   }
 
+  function findClosestErrorBoundary(fiber : Fiber): ?Fiber {
+    let maybeErrorBoundary = fiber.return;
+    while (maybeErrorBoundary) {
+      if (maybeErrorBoundary.tag === ClassComponent) {
+        const instance = maybeErrorBoundary.stateNode;
+        if (typeof instance.unstable_handleError === 'function') {
+          return maybeErrorBoundary;
+        }
+      }
+      maybeErrorBoundary = maybeErrorBoundary.return;
+    }
+    return null;
+  }
+
+  function handleError(failedUnitOfWork : Fiber, error : any) {
+    const errorBoundary = findClosestErrorBoundary(failedUnitOfWork);
+    if (errorBoundary) {
+      handleErrorInBoundary(errorBoundary, error);
+      return;
+    }
+    // TODO: Do we need to reset nextUnitOfWork here?
```

Comment:
It is always safe to set nextUnitOfWork to `null` because it will just find its place again.